src/security/acl_api.c

```diff
@@ -372,15 +397,25 @@ is_in_ace_list(uint8_t *addr, struct daos_acl *acl)
 	return addr >= start_addr && addr < end_addr;
 }
 
+static bool
+is_first_ace(struct daos_acl *acl, struct daos_ace *ace)
```

Comment:
Just for your consideration; not a change request: the prototype of this helper function implies (to me):

- I need to pass non-NULL acl and non-NULL ace.
- It'll check if ace is the first entry of acl by comparing their contents.

That surely isn't your intention. So this `daoc_acl_get_next_ace`-specific check might be better coded directly around line 415.

Reply:
Agreed. Going with your suggestion.